src/maxminddb.c

```diff
@@ -394,15 +400,22 @@ LOCAL int read_metadata(MMDB_s *mmdb)
         .offset = 0
     };
 
-    mmdb->metadata.node_count =
-        value_for_key_as_uint32(&metadata_start, "node_count");
+    int status =
+        value_for_key_as_uint32(&metadata_start, "node_count",
+                                &mmdb->metadata.node_count);
+    if (MMDB_SUCCESS != status) {
+        return status;
+    }
     if (!mmdb->metadata.node_count) {
         DEBUG_MSG("could not find node_count value in metadata");
         return MMDB_INVALID_METADATA_ERROR;
     }
 
-    mmdb->metadata.record_size =
-        value_for_key_as_uint16(&metadata_start, "record_size");
+    status = value_for_key_as_uint16(&metadata_start, "record_size",
```

Comment:
@autarch, it turns out that before the 2014-01-28 GeoIP2 City release, `record_size` was a `uint32`. I am pretty sure that such databases would have been broken before this change on big endian machines (as reading the union type as `uint16` wouldn't work there). Thoughts on how we should proceed? At the very least, it seems like we should warn people somehow.